#### Exercicis de Web Scraping.

### - Exercici 1
#### Realitza web scraping de dues de les tres pàgines web proposades utilitzant BeautifulSoup primer i Selenium després. 
#### - http://quotes.toscrape.com
#### - https://www.bolsamadrid.es
#### - www.wikipedia.es (fes alguna cerca primer i escrapeja algun contingut)

El web scraping és una tècnica que s'utilitza per a extreure informació de pàgines web. És una tècnica molt comuna en la mineria de dades i pot utilitzar-se per obtenir informació de qualsevol lloc web que tingui dades públiques.

BeautifulSoup i Selenium són dues eines per realitzar web scraping. BeautifulSoup és útil per extreure dades de pàgines web que no requereixen interacció, mentre que Selenium s'utilitza per interactuar amb llocs web més complexos que requereixen una interacció més avançada.
Per a cada una de les pàgines web, primer farem web scraping amb BeautifulSoup per obtenir les dades, i després ho farem amb Selenium per veure les diferències en les dades obtingudes.

#### - http://quotes.toscrape.com

In [403]:
import pandas as pd
import numpy as np
import csv
import matplotlib.pyplot as plt

In [443]:
import requests
from bs4 import BeautifulSoup

url = "http://quotes.toscrape.com"
response = requests.get(url)
soup = BeautifulSoup(response.content, "html.parser")

Aquest codi emet una sol·licitud HTTP GET a l'URL donat. Recupera les dades HTML que el servidor envia i emmagatzema aquestes dades en un objecte Python. Aquest és un codi HTML que mostra cometes, els seus autors i etiquetes. Cada cita està continguda dins d'un div amb la classe "quote" i conté el text de la cita, l'autor i un conjunt d'etiquetes. El text de la cita està contingut dins d'un interval amb la classe "text" i l'autor està contingut dins d'un petit element amb la classe "autor". Les etiquetes estan contingudes dins d'un div amb la classe "etiquetes" i cada etiqueta està continguda dins d'un element d'ancoratge amb la classe "etiqueta". Les etiquetes també contenen un metaelement amb la classe "keywords" que enumera les paraules clau associades a la cita.
El codi també inclou enllaços a les pàgines de l'autor, que tenen el format "/author/Author-Name".

Beautiful Soup és una biblioteca de Python per analitzar dades estructurades. Us permet interactuar amb HTML d'una manera similar a com interactueu amb una pàgina web mitjançant eines de desenvolupament. La biblioteca exposa un parell de funcions intuïtives que podeu utilitzar per explorar l'HTML que heu rebut.

Cerca elements per ID
En una pàgina web HTML, cada element pot tenir assignat un atribut id. Com el nom ja indica, aquest atribut id fa que l'element es pugui identificar de manera única a la pàgina. 

In [444]:
quote = soup.find("span", class_="text").text
print(quote)

“The world as we have created it is a process of our thinking. It cannot be changed without changing our thinking.”


El codi ens ha imprès la primera cita del lloc web

In [406]:
# Busca tots els elements HTML que contenen els noms dels autors
author_tags = soup.find_all("small", class_="author")

# Recorre la llista d'elements i extreu el text dels noms d'autors
authors = [author_tag.text for author_tag in author_tags]

# Imprimeix la llista d'autors
print(authors)

['Albert Einstein', 'J.K. Rowling', 'Albert Einstein', 'Jane Austen', 'Marilyn Monroe', 'Albert Einstein', 'André Gide', 'Thomas A. Edison', 'Eleanor Roosevelt', 'Steve Martin']


Aquest codi ens ha buscat tots els elements HTML que contenen la classe "author" i són de tipus "small", i extreura el text dels noms d'autors que es troben dins d'aquests elements. El resultat es una llista de noms d'autors.

In [407]:
# Busca l'element HTML que conté les etiquetes "tag"
tags_container = soup.find("div", class_="tags")

# Busca tots els enllaços HTML dins de l'element "tags"
tag_links = tags_container.find_all("a", class_="tag")

# Recorre la llista d'enllaços i extreu el text de les etiquetes
tags = [tag_link.text for tag_link in tag_links]

# Imprimeix la llista d'etiquetes
print(tags)

['change', 'deep-thoughts', 'thinking', 'world']


Hem impres una llista amb els noms de les etiquetes "tag" que apareixen a la pàgina web.

In [408]:
tag_name = 'thinking'
tag_url = f"{url}/tag/{tag_name}"
tag_response = requests.get(tag_url)
tag_soup = BeautifulSoup(tag_response.content, "html.parser")
quotes = tag_soup.find_all("span", class_="text")

for quote in quotes:
    print(quote.text)

“The world as we have created it is a process of our thinking. It cannot be changed without changing our thinking.”
“The trouble with having an open mind, of course, is that people will insist on coming along and trying to put things in it.”


Hem imprès totes les citacions que tenen la paraula "thinking".

In [409]:
tags = soup.find_all("a", class_="tag")
for tag in tags:
    tag_url = url + tag["href"]
    tag_response = requests.get(tag_url)
    tag_soup = BeautifulSoup(tag_response.content, "html.parser")
    quotes = tag_soup.find_all("span", class_="text")
    for quote in quotes:
        author = tag_soup.find("small", class_="author").text
        if "Marilyn Monroe" in author:
            print(quote.text)

“Imperfection is beauty, madness is genius and it's better to be absolutely ridiculous than absolutely boring.”
“This life is what you make it. No matter what, you're going to mess up sometimes, it's a universal truth. But the good part is you get to decide how you're going to mess it up. Girls will be your friends - they'll act like it anyway. But just remember, some come, some go. The ones that stay with you through everything - they're your true best friends. Don't let go of them. Also remember, sisters make the best friends in the world. As for lovers, well, they'll come and go too. And baby, I hate to say it, most of them - actually pretty much all of them are going to break your heart, but you can't give up because if you give up, you'll never find your soulmate. You'll never find that half who makes you whole and that goes for everything. Just because you fail once, doesn't mean you're gonna fail at everything. Keep trying, hold on, and always, always, always believe in yourself

Hem buscat totes les etiquetes "a" amb la classe "tag",  el vincle per obtenir les pàgines que contenen les cites corresponents i, per cada pàgina, busca les frases amb l'etiqueta "span" i la classe "text". Després comprova si l'autor de la cita és "Marilyn Monroe" i, si és així, imprimeix la frase.

SELENIUM

In [410]:
from selenium.webdriver.chrome.service import Service

service = Service('/usr/local/bin/chromedriver')
service.start()

In [411]:
from selenium import webdriver
from selenium.webdriver.common.by import By

url= "http://quotes.toscrape.com"
driver = webdriver.Chrome()  # o el driver que vulguis utilitzar
driver.get(url)
quote = driver.find_elements(By.XPATH, "//span[@class='text']")[0].text
print(quote)

“The world as we have created it is a process of our thinking. It cannot be changed without changing our thinking.”


In [412]:
links = [link.get_attribute("href") for link in driver.find_elements(By.XPATH, "//a[@href]")]
print(links)

['http://quotes.toscrape.com/', 'http://quotes.toscrape.com/login', 'http://quotes.toscrape.com/author/Albert-Einstein', 'http://quotes.toscrape.com/tag/change/page/1/', 'http://quotes.toscrape.com/tag/deep-thoughts/page/1/', 'http://quotes.toscrape.com/tag/thinking/page/1/', 'http://quotes.toscrape.com/tag/world/page/1/', 'http://quotes.toscrape.com/author/J-K-Rowling', 'http://quotes.toscrape.com/tag/abilities/page/1/', 'http://quotes.toscrape.com/tag/choices/page/1/', 'http://quotes.toscrape.com/author/Albert-Einstein', 'http://quotes.toscrape.com/tag/inspirational/page/1/', 'http://quotes.toscrape.com/tag/life/page/1/', 'http://quotes.toscrape.com/tag/live/page/1/', 'http://quotes.toscrape.com/tag/miracle/page/1/', 'http://quotes.toscrape.com/tag/miracles/page/1/', 'http://quotes.toscrape.com/author/Jane-Austen', 'http://quotes.toscrape.com/tag/aliteracy/page/1/', 'http://quotes.toscrape.com/tag/books/page/1/', 'http://quotes.toscrape.com/tag/classic/page/1/', 'http://quotes.toscra

In [413]:
love_quotes = driver.find_elements(By.XPATH, "//span[@class='text' and contains(text(), 'love')]")
for quote in love_quotes:
    print(quote.text)

“It is better to be hated for what you are than to be loved for what you are not.”


Hem obtingut la primera cita de la pàgina web, els enllaços de la web i també les cita que conte la paraula "love".

https://es.wikipedia.org/wiki/Andy_Warhol

In [414]:
url1 = "https://es.wikipedia.org/wiki/Andy_Warhol"
page = requests.get(url1)
html= page.content
soup_warhol = BeautifulSoup(html, "html.parser")

# Troba el primer div amb l'identificador "mw-content-text"
content_div = soup_warhol.find("div", id="mw-content-text")

# Troba el primer paràgraf dins del div
first_paragraph = content_div.find("p")

# Imprimeix el text del paràgraf
print(first_paragraph.text)

Andrew Warhol (Pittsburgh, 6 de agosto de 1928-Nueva York, 22 de febrero de 1987) fue un artista plástico y actor estadounidense que desempeñó un papel crucial en el nacimiento y desarrollo del pop art. Tras una exitosa carrera como ilustrador profesional, Warhol adquirió fama mundial por su trabajo en pintura, cine de vanguardia y literatura, notoriedad que vino respaldada por una hábil relación con los medios y por su rol como gurú de la modernidad. Warhol actuó como enlace entre artistas e intelectuales, pero también entre aristócratas, homosexuales, celebridades de Hollywood, modelos, bohemios y pintorescos personajes urbanos. 



In [416]:
title = soup_warhol.title.string
print(title)

Andy Warhol - Wikipedia, la enciclopedia libre


In [417]:
velvet_section = soup_warhol.find('span', id='Música')

# Extreure la informació rellevant d'aquesta secció
info = velvet_section.find_next('p').get_text()

print(info)

A mediados de los sesenta, Warhol «adoptó» a una banda llamada The Velvet Underground, y les convirtió en un elemento crucial de la exposición de arte multimedia que organizaba, la Exploding Plastic Inevitable. Warhol convirtió a Paul Morrissey en el agente de la banda, y les presentó a Nico, quien se convertiría en miembro de la banda a petición de Warhol. En 1966 produjo su primer disco, titulado "The Velvet Underground and Nico", y realizó la decoración del disco. En realidad, la producción de Warhol se limitó a pagar por el estudio de grabación. Tras la publicación de este álbum, Warhol y el líder de la banda, Lou Reed, empezaron a disentir sobre la dirección que la banda debería tomar. Esta tensión pudo haber causado el final de su buena relación artística.



SELENIUM 

In [438]:
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options as ChromeOptions
import time

op = webdriver.ChromeOptions()
op.add_argument('--enable-extensions')

wikiSelenium = webdriver.Chrome(options=op)
url1= "https://es.wikipedia.org/wiki/Andy_Warhol"
wikiSelenium.get(url1)
time.sleep(5)

# Obtenir informació del web, en aquest cas el títol de la  web
title = wikiSelenium.title
print(title)

# Busca l'enllaç a la secció "La Factory de Nueva York"
element = wikiSelenium.find_element(By.XPATH, "//a[@href='#La_Factory_de_Nueva_York']")

# Obté el contingut de l'enllaç
contingut_enllac = element.text

# Imprimeix el contingut de l'enllaç
print(contingut_enllac)
wikiSelenium.quit()

Andy Warhol - Wikipedia, la enciclopedia libre
3.6 La Factory de Nueva York


In [434]:
# Inicialitzar el driver de selenium
driver = webdriver.Chrome('./chromedriver')

#    web
driver.get("https://es.wikipedia.org/wiki/Andy_Warhol")

time.sleep(10)

ul = driver.find_element("xpath", '//*[@id="mw-content-text"]')
data = []

# Buscar els enllaços dins de la llista y mostrar informació
enlaces = ul.find_elements(By.TAG_NAME, "a")
for enlace in enlaces:
    title = enlace.get_attribute("title")
    url = "https://es.wikipedia.org/wiki/Andy_Warhol" + enlace.get_attribute("href")
    text = enlace.text
    data.append([title, url])

    # Imprimir 
    print("Títol: ", title)
    print("URL: ", url)
    print("Text: ", text)
    print()
# Tancar el driver de selenium
driver.quit()

/var/folders/fn/wxghhlc100ncy9y8vb1979kw0000gn/T/ipykernel_36177/1934861561.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome('./chromedriver')


Títol:  Andy Warhol (canción)
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol_(canci%C3%B3n)
Text:  Andy Warhol (canción)

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Archivo:Andy_Warhol_at_the_Jewish_Museum_(by_Bernard_Gotfryd)_%E2%80%93_LOC.jpg
Text:  

Títol:  Bandera de Estados Unidos
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Archivo:Flag_of_the_United_States.svg
Text:  

Títol:  Pittsburgh
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Pittsburgh
Text:  Pittsburgh

Títol:  Estados Unidos
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Estados_Unidos
Text:  Estados Unidos

Títol:  Bandera de Estados Unidos
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Archivo:Flag_of_the_United_States.svg
Text:  

Títol:  Nueva York
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wiki

Títol:  Mary Harron
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Mary_Harron
Text:  Mary Harron

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#Biograf%C3%ADa
Text:  1 Biografía

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#Infancia
Text:  1.1 Infancia

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#Principios_de_su_carrera
Text:  1.2 Principios de su carrera

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#A%C3%B1os_1960
Text:  1.3 Años 1960

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#Atentado
Text:  1.4 Atentado

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#A%C3%B1os_1970
Text:  1.5 Años 1970

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Wa

Títol:  Díptico de Marilyn
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/D%C3%ADptico_de_Marilyn
Text:  Díptico de Marilyn

Títol:  Latas de sopa Campbell
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Latas_de_sopa_Campbell
Text:  Latas de sopa Campbell

Títol:  100 botellas de cola (aún no redactado)
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/w/index.php?title=100_botellas_de_cola&action=edit&redlink=1
Text:  100 botellas de cola

Títol:  100 billetes de dólar (aún no redactado)
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/w/index.php?title=100_billetes_de_d%C3%B3lar&action=edit&redlink=1
Text:  100 billetes de dólar

Títol:  John Giorno
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/John_Giorno
Text:  John Giorno

Títol:  Coca-Cola
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Coca-Cola
Text:  Coca-Cola

Tít

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_note-popism287-295-14
Text:  14

Títol:  Asesinato
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Asesinato
Text:  asesinato

Títol:  Robert F. Kennedy
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Robert_F._Kennedy
Text:  Robert F. Kennedy

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_note-stiles1996-15
Text:  15

Títol:  Editar sección: Años 1970
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/w/index.php?title=Andy_Warhol&action=edit&section=6
Text:  editar

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Archivo:BMW_M1_Art_Car.jpg
Text:  

Títol:  Aumentar
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Archivo:BMW_M1_Art_Car.jpg
Text:  

Títol:  BMW M1
URL:  https://es.wiki

Títol:  Editar sección: Muerte
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/w/index.php?title=Andy_Warhol&action=edit&section=8
Text:  editar

Títol:  Vesícula biliar
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Ves%C3%ADcula_biliar
Text:  vesícula

Títol:  Arritmia
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Arritmia
Text:  arritmia

Títol:  Hiponatremia
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Hiponatremia
Text:  intoxicación de agua

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_note-22
Text:  22

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Archivo:Warhol_grave.jpg
Text:  

Títol:  Aumentar
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Archivo:Warhol_grave.jpg
Text:  

Títol:  Cementerio
URL:  https://es.wikipedia.org/wiki/Andy_W

Títol:  Monotipia
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Monotipia
Text:  monotipia

Títol:  Publicidad
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Publicidad
Text:  publicidad

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_note-32
Text:  32

Títol:  Papa
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Papa
Text:  Papa

Títol:  Dibujos animados
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Dibujos_animados
Text:  dibujos animados

Títol:  Televisión
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Televisi%C3%B3n
Text:  televisivos

Títol:  Expresionismo abstracto
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Expresionismo_abstracto
Text:  expresionismo abstracto

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_note-41
Text:  41

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_note-42
Text:  42

Títol:  Viva (actriz) (aún no redactado)
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/w/index.php?title=Viva_(actriz)&action=edit&redlink=1
Text:  Viva

Títol:  Paul Morrissey
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Paul_Morrissey
Text:  Paul Morrissey

Títol:  Serie-B (aún no redactado)
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/w/index.php?title=Serie-B&action=edit&redlink=1
Text:  serie-B

Títol:  Drácula
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Dr%C3%A1cula
Text:  Drácula

Títol:  Víctor Frankenstein
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/V%C3%ADctor_Frankenstein
Text:  Frankens

Títol:  Televisión
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Televisi%C3%B3n
Text:  televisivo

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_note-48
Text:  48

Títol:  Moda
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Moda
Text:  moda

Títol:  Edie Sedgwick
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Edie_Sedgwick
Text:  Edie Sedgwick

Títol:  Modisto
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Modisto
Text:  diseñadora de moda

Títol:  Halston
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Halston
Text:  Halston

Títol:  Performance
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Performance
Text:  Performance

Títol:  Diapositiva
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Diapositiva
Text:  diapositivas



Títol:  Aumentar
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Archivo:Andy_Warhol_and_Ulli_Lommel_on_set_of_Cocaine_Cowboys.jpg
Text:  

Títol:  Ulli Lommel
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Ulli_Lommel
Text:  Ulli Lommel

Títol:  Cocaine Cowboys
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Cocaine_Cowboys
Text:  Cocaine Cowboys

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_note-61
Text:  61

Títol:  Crispin Glover
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Crispin_Glover
Text:  Crispin Glover

Títol:  Oliver Stone
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Oliver_Stone
Text:  Oliver Stone

Títol:  The Doors (película)
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/The_Doors_(pel%C3%ADcula)
Text:  The Doors

Títol:  David Bowie

Títol:  Secker and Warburg
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Secker_and_Warburg
Text:  Secker and Warburg

Títol:  ISBN
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/ISBN
Text:  ISBN

Títol:  Especial:FuentesDeLibros/0-436-28015-9
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Especial:FuentesDeLibros/0-436-28015-9
Text:  0-436-28015-9

Títol:  OCLC
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/OCLC
Text:  OCLC

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://www.worldcat.org/oclc/50215773
Text:  50215773

Títol:  Ayuda:Errores en las referencias
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Ayuda:Errores_en_las_referencias#deprecated_params
Text:  

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-3
Text:  ↑

Títol:  Volver arriba
URL: 

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-Cueva_12-1
Text:  b

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://web.archive.org/web/20121130155904/http://www.ciudaddemujeres.com/articulos/Cuarenta-anos-del-grito
Text:  «¿Quién era Valérie Solanas?»

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttp://www.ciudaddemujeres.com/articulos/Cuarenta-anos-del-grito
Text:  el original

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-Harding_13-0
Text:  ↑

Títol:  TDR (journal) (aún no redactado)
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/w/index.php?title=TDR_(journal)&action=edit&redlink=1
Text:  TDR/The Drama Review

Títol:  Digital object identifier
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Digital_object_identifier
Text:  doi

Títol:  
URL:  https://es.wikipedia.org/w

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-21
Text:  ↑

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttp://andrechahil.com/wien-1985-phaenomen-fax-art-beuys-warhol-und-higashiyama-setzen-dem-kalten-krieg-ein-zeichen
Text:  Wien 1985: Phänomen Fax-Art. Beuys, Warhol und Higashiyama setzen dem Kalten Krieg ein Zeichen.

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-22
Text:  ↑

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttp://query.nytimes.com/gst/fullpage.html?res=9B0DE3DA1639F930A25757C0A961948260
Text:  «Hospital Asserts it Gave Warhol Adequate Care»

Títol:  The New York Times
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/The_New_York_Times
Text:  The New York Times

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cit

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-32
Text:  ↑

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-33
Text:  ↑

Títol:  Marcel Duchamp
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Marcel_Duchamp
Text:  Marcel Duchamp

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-34
Text:  ↑

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttp://artsalesindex.artinfo.com/artsalesindex/asi/lots/10388409
Text:  «Auction Results: Andy Warhol's Campbell's Soup Can»

Títol:  Louise Blouin Media (aún no redactado)
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/w/index.php?title=Louise_Blouin_Media&action=edit&redlink=1
Text:  Louise Blouin Media

Títol:  Ayuda:Cómo recuperar un enlace roto
URL:  https://es.wikipedia.org

Títol:  New York City
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/New_York_City
Text:  New York City

Títol:  ISBN
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/ISBN
Text:  ISBN

Títol:  Especial:FuentesDeLibros/0-8109-1761-0
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Especial:FuentesDeLibros/0-8109-1761-0
Text:  0-8109-1761-0

Títol:  OCLC
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/OCLC
Text:  OCLC

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://www.worldcat.org/oclc/19389231
Text:  19389231

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-365_Takes_45-0
Text:  ↑

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://archive.org/details/andywarhol365tak0000unse
Text:  Andy Warhol: 365 Takes

Títol:  New York City
URL:  https://es.wikipedia.org/wiki/Andy_Warh

Títol:  Jon Pareles
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Jon_Pareles
Text:  Pareles, Jon

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttp://query.nytimes.com/gst/fullpage.html?res=950DEED71238F932A35751C1A96F948260
Text:  «Review/Rock; 'Songs for Drella,' A Tribute to Warhol»

Títol:  The New York Times
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/The_New_York_Times
Text:  The New York Times

Títol:  Volver arriba
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Andy_Warhol#cite_ref-56
Text:  ↑

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://web.archive.org/web/20080528171637/http://www.rollingstone.com/artists/loureed/albums/album/198555/review/5944531/songs_for_drella
Text:  «Lou Reed: Songs for Drella»

Títol:  Rolling Stone
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Rolling_Stone
Text:  Rolling Stone

Títol:  
URL:  http

Títol:  New York City
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/New_York_City
Text:  New York City

Títol:  Continuum International Publishing Group
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Continuum_International_Publishing_Group
Text:  Continuum International Publishing Group

Títol:  ISBN
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/ISBN
Text:  ISBN

Títol:  Especial:FuentesDeLibros/0-8264-1334-X
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Especial:FuentesDeLibros/0-8264-1334-X
Text:  0-8264-1334-X

Títol:  José Esteban Muñoz
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Jos%C3%A9_Esteban_Mu%C3%B1oz
Text:  José Esteban Muñoz

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://archive.org/details/loneratballlifeo0000guil
Text:  Loner at the Ball: The Life of Andy Warhol

Títol:  Princeton University Press
U

Títol:  wikidata:Q5603
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://www.wikidata.org/wiki/Q5603
Text:  Q5603

Títol:  Commonscat
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Wikimedia_Commons
Text:  

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://commons.wikimedia.org/wiki/Category:Andy_Warhol
Text:  Andy Warhol

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://commons.wikimedia.org/wiki/Special:MediaSearch?type=image&search=%22Q5603%22
Text:  Q5603

Títol:  Wikiquote
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Wikiquote
Text:  

Títol:  q:Andy Warhol
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikiquote.org/wiki/Andy_Warhol
Text:  Andy Warhol

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://www.worldcat.org/identities/containsVIAFID/56617645
Text:  WorldCat

Títol:  Fichero de Autoridades Virtual Internacional
URL:  https://es.wikipedia.org/wiki/And

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://americanart.si.edu/collections/search/artist/?id=5233
Text:  5233

Títol:  Internet Movie Database
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Internet_Movie_Database
Text:  IMDb

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://wikidata-externalid-url.toolforge.org/?p=345&url_prefix=https://www.imdb.com/&id=nm0912238
Text:  nm0912238

Títol:  Wikidata
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Wikidata
Text:  

Títol:  wikidata:Q5603
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://www.wikidata.org/wiki/Q5603
Text:  

Títol:  Commonscat
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://es.wikipedia.org/wiki/Wikimedia_Commons
Text:  

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://commons.wikimedia.org/wiki/Category:Andy_Warhol
Text:  

Títol:  
URL:  https://es.wikipedia.org/wiki/Andy_Warholhttps://commons.wikimed

In [437]:
enlaces

[<selenium.webdriver.remote.webelement.WebElement (session="8e050b5cc184f2c95d558867fdc034c5", element="fdce1319-0272-4795-9121-a38892ad9287")>,
 <selenium.webdriver.remote.webelement.WebElement (session="8e050b5cc184f2c95d558867fdc034c5", element="25826964-dfe1-423e-95ad-1f0a85148a38")>,
 <selenium.webdriver.remote.webelement.WebElement (session="8e050b5cc184f2c95d558867fdc034c5", element="12e13138-c4ec-4890-bef6-28bd100e67a7")>,
 <selenium.webdriver.remote.webelement.WebElement (session="8e050b5cc184f2c95d558867fdc034c5", element="bc19fb7e-926c-48f1-a4a1-3ff83680602f")>,
 <selenium.webdriver.remote.webelement.WebElement (session="8e050b5cc184f2c95d558867fdc034c5", element="c5c5ede8-dafd-41e9-8aed-32d41c51ee7c")>,
 <selenium.webdriver.remote.webelement.WebElement (session="8e050b5cc184f2c95d558867fdc034c5", element="76543748-1eeb-43bc-a5cf-1eb08beecf4c")>,
 <selenium.webdriver.remote.webelement.WebElement (session="8e050b5cc184f2c95d558867fdc034c5", element="98a37980-7b3e-40b1-bee1-16

In [441]:
# Convertir la lista en un dataframe de Pandas
wikipedia = pd.DataFrame(data, columns=['Títol', 'URL'])

# Imprimir el dataframe
wikipedia.head(20)

Títol  \
0                   Andy Warhol (canción)   
1                                           
2               Bandera de Estados Unidos   
3                              Pittsburgh   
4                          Estados Unidos   
5               Bandera de Estados Unidos   
6                              Nueva York   
7                          Estados Unidos   
8           Trastornos del ritmo cardíaco   
9   Ver y modificar los datos en Wikidata   
10  Ver y modificar los datos en Wikidata   
11                         Estados Unidos   
12                            Catolicismo   
13  Ver y modificar los datos en Wikidata   
14  Ver y modificar los datos en Wikidata   
15  Ver y modificar los datos en Wikidata   
16                            John Giorno   
17  Ver y modificar los datos en Wikidata   
18       Carnegie Institute of Technology   
19            Universidad Carnegie Mellon   

                                                  URL  
0   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
1   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
2   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
3   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
4   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
5   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
6   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
7   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
8   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
9   https://es.wikipedia.org/wiki/Andy_Warholhttps...  
10  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
11  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
12  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
13  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
14  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
15  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
16  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
17  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
18  https://es.wikipedia.org/wiki/Andy_Warholhttps...  
19  https://es.wikipedia.org/wiki/Andy_Warholhttps...

### - Exercici 2
#### Documenta en un Word el teu conjunt de dades generat amb la informació que tenen els diferents arxius de Kaggle.

In [442]:
wikipedia.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1023 entries, 0 to 1022
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Títol   1023 non-null   object
 1   URL     1023 non-null   object
dtypes: object(2)
memory usage: 16.1+ KB


# Informació general

El conjunt de dades és informació sobre Andy Warhol, va ser un artista plàstic i actor nord-americà que va exercir un paper crucial en el naixement i desenvolupament del pop art. Després d'una carrera d'èxit com a il·lustrador professional, Warhol va adquirir fama mundial pel seu treball en pintura, cinema d'avantguarda i literatura, notorietat que va venir recolzada per una hàbil relació amb els mitjans i pel seu rol com a guru de la modernitat.
La informació l'hem obtingut de: https://es.wikipedia.org/wiki/Andy_Warhol"
El conjunt de dades és tots els links que hi han a la informació D'Andy Warhol. Les dades són proporcionat per Wikipedia en espanyol, un projecte d'enciclopèdia web multilingüe de contingut gratuït basat en un model d'edició oberta. La Wikipèdia creix cada dia gràcies a la participació de persones d'arreu del món, sent el projecte de recopilació de coneixement més gran que s'ha dut a terme en la història de la humanitat.

# Contingut dels arxius

2 columnes i 1023 files, les dues columnes son type object.
Les columnes del conjunt de dades són les següents:

- Títol (informació dels links)
- URL (l'adreça web del link)


# Referències

https://realpython.com/modern-web-automation-with-python-and-selenium/
https://www.selenium.dev/documentation/webdriver/getting_started/
https://selenium-python.readthedocs.io/getting-started.html#simple-usage
https://es.wikipedia.org/wiki/Andy_Warhol

# Llicència

Aquesta pàgina es va editar per última vegada el 6 mar 2023 a les 14:07.
El text està disponible sota la Llicència Creative Commons Atribució Compartir Igual 3.0; es poden aplicar clàusules addicionals. En utilitzar aquest lloc, acceptem els  termes d'ús i la política de privadesa.
Wikipedia® és una marca registrada de la Fundació Wikimedia, Inc., una organització sense ànim de lucre.